sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncClient.java

```diff
@@ -918,11 +921,34 @@ private TableAccessPolicy toTableAccessPolicy(AccessPolicy accessPolicy) {
     Mono<Response<Void>> setAccessPoliciesWithResponse(List<TableSignedIdentifier> tableSignedIdentifiers,
                                                        Context context) {
         context = context == null ? Context.NONE : context;
+        List<SignedIdentifier> signedIdentifiers = null;
+
+        /*
+        We truncate to seconds because the service only supports nanoseconds or seconds, but doing an
+        OffsetDateTime.now will only give back milliseconds (more precise fields are zeroed and not serialized). This
+        allows for proper serialization with no real detriment to users as sub-second precision on active time for
+        signed identifiers is not really necessary.
+         */
+        if (tableSignedIdentifiers != null) {
+            signedIdentifiers = tableSignedIdentifiers.stream()
+                .map(this::toSignedIdentifier)
+                .collect(Collectors.toList());
+
+            for (SignedIdentifier identifier : signedIdentifiers) {
+                if (identifier.getAccessPolicy() != null && identifier.getAccessPolicy().getStart() != null) {
+                    identifier.getAccessPolicy().setStart(
+                        identifier.getAccessPolicy().getStart().truncatedTo(ChronoUnit.SECONDS));
+                }
+                if (identifier.getAccessPolicy() != null && identifier.getAccessPolicy().getExpiry() != null) {
+                    identifier.getAccessPolicy().setExpiry(
+                        identifier.getAccessPolicy().getExpiry().truncatedTo(ChronoUnit.SECONDS));
+                }
```

Comment:
This could be made part of the `Stream.map` above to prevent iterating the collection twice